src/main/java/io/confluent/connect/hdfs/tools/SchemaSourceTask.java

```diff
@@ -99,24 +120,25 @@ public void start(Map<String, String> props) {
           .put("id", id)
           .put("seqno", seqno);
 
-      for (Field field: valueSchema.fields()) {
-        log.debug("Field: {}", field.name());
-        log.debug("Value: {}", data.get(field.name()));
-      }
       Map<String, Long> ccOffset = Collections.singletonMap(SEQNO_FIELD, seqno);
-      SourceRecord srcRecord = new SourceRecord(partition, ccOffset, topic, 0, Schema.STRING_SCHEMA, "key", valueSchema, data);
-      log.debug("Source record: {}", srcRecord.toString());
+      SourceRecord srcRecord = new SourceRecord(partition, ccOffset, topic, id, Schema.STRING_SCHEMA, "key", valueSchema, data);
+      System.out.println("{\"task\": " + id + ", \"seqno\": " + seqno + "}");
       List<SourceRecord> result = Arrays.asList(srcRecord);
       seqno++;
       count++;
       return result;
     } else {
+      synchronized (this) {
+        this.wait();
```

Comment:
Yeah, the idea is that we should just sleep because there's no work to do (and sleep indefinitely because there will never be any more work to do). I think we actually wanted this before any of the other additions I made. Without it, the task will basically just busy loop by returning empty lists immediately for every iteration.